olive/workflows/run/run.py

```diff
@@ -10,8 +10,6 @@
 from pathlib import Path
 from typing import List, Union
 
-import onnxruntime as ort
```

Comment:
Or we will need to restructure the onnxruntime imports in other parts of the codebase to make it lazy.

Reply:
Since onnxruntime is not a dependency of Olive, if the user env doesn't have onnxruntime, I expect onnxruntime should be installed by running `--setup` command. This PR is to fix a bug that I met before: I was trying to install correct onnxruntime package by running `--setup` command, but onnxruntime was used before running the setup (before installing the onnxruntime).

Reply:
onnxruntime is a dependency of olive. it's just that we don't install it directly since there are many ort packages. Even the installation instructions say it https://microsoft.github.io/Olive/getstarted/installation.html 

This won't resolve your bug since the `from olive.passes import Pass` at the top which is needed for pass registration and validation triggers `onnxruntime` anyways.

Reply:
This is similar to how transformers requires one of torch or tensorflow to work but doesn't install them as a direct dependency.

Reply:
I see I got your point. If we expect the users who install olive without extra to install onnxruntime manually, I think we can improve our warning/error message by checking onnxuntime import at the entrance of run method.

Reply:
Yes, I agree. Another option is to make onnxruntime imports in all passes lazy. I just checked and some passes import them lazy while others import them at the top of the file.

Reply:
I also did a similar check. Since we still have openvino and snpe support, I think we can make all onnxruntime imports lazy. I will check this later,